src/Featurizers/ForecastingPivotFeaturizer.h

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "../Archive.h"
+#include "../Traits.h"
+
+#include "Components/InferenceOnlyFeaturizerImpl.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         ForecastingPivotTransformer
+///  \brief         Expand output from upstream featurizers where all values
+///                 in the same column across matrixes are not null/empty
+///
+template <typename InputT>
+class ForecastingPivotTransformer :
+    public Components::InferenceOnlyTransformerImpl<
+        typename std::vector<Microsoft::Featurizer::RowMajMatrix<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>>,
+        std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>
+    > {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType =
+        Components::InferenceOnlyTransformerImpl<
+            typename std::vector<Microsoft::Featurizer::RowMajMatrix<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>>,
+            std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>
+        >;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    ForecastingPivotTransformer(void) = default;
+    ForecastingPivotTransformer(Archive &ar);
+
+    ~ForecastingPivotTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(ForecastingPivotTransformer);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Types
+    // |
+    // ----------------------------------------------------------------------
+    using NullableType                      = typename Microsoft::Featurizer::Traits<InputT>::nullable_type;
+    using MatrixType                        = Microsoft::Featurizer::RowMajMatrix<NullableType>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+
+    // MSVC runs into problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        if (input.empty()) {
+            return;
+        }
+        
+        std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type> ret;
```

Comment:
This should also just be InputT. We don't need the nullable type here either.

Reply:
Good catch! Changed to InputT.